core/src/main/java/com/google/zxing/oned/OneDimensionalCodeWriter.java

```diff
@@ -90,6 +92,9 @@ private static BitMatrix renderResult(boolean[] code, int width, int height, int
     return output;
   }
 
+  protected static boolean checkNumeric(String contents) {
+    return NUMERIC.matcher(contents).matches();
```

Comment:
Why not just have this method check whether it matches and throw the exception? that way you don't have to repeat the check and throw everywhere. It can be `final` too.